core/src/main/java/brooklyn/util/flags/TypeCoercions.java

```diff
@@ -709,6 +710,13 @@ public QuorumCheck apply(final String input) {
                 return QuorumChecks.of(input);
             }
         });
+        registerAdapter(ArrayList.class, String[].class, new Function<ArrayList, String[]>() {
```

Comment:
Could we declare this as `List.class`, rather than `ArrayList.class`? Or does it need to be the concrete type?